### PR TITLE
Add man command

### PR DESCRIPTION
**Pull request by Merovius**

### CODE REVIEW

File: manpages.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var manpagesMatcher = regexp.MustCompile(`\b(\w+)\((\d+)\)(\W|$)`)
```
Comment (stapelberg) [inline manpages.go]: Strictly speaking, there are sections which aren’t strictly numerical, e.g. https://manpages.debian.org/jessie/libedit-dev/editline.3edit. A more practical example is 3pl. That said, I expect that in practice, references will usually be restricted to numerical sections. So, maybe add a TODO?

File: manpages.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+func runnerManpages(parsed Message) {
+	for _, l := range extractManpages(parsed.Trailing) {
+		Privmsg(Target(parsed), "[manpage] "+l)
+	}
+}
+
+func extractManpages(msg string) (links []string) {
+	msg = strings.TrimPrefix(msg, "frank: ")
```
Comment (stapelberg) [inline manpages.go]: I can easily see people writing e.g. “man cherti”. Maybe we should require the “frank: ” prefix?

**Review by stapelberg — COMMENTED** [review]
Looks good overall.

File: manpages.go (hunk 1 above)
Comment (Merovius) [reply manpages.go]: Is there an example of a section not matching "\d[a-z]+"? I tried to find the most restrictive pattern that still matches everything important to minimize false-positives.

File: manpages.go (hunk 2 above)
Comment (Merovius) [reply manpages.go]: Grepping through my IRC-logs I indeed find two false positives. Fixed.

File: manpages.go (hunk 1 above)
Comment (stapelberg) [reply manpages.go]: Yes: section names use dashes and underscores. `[0-9][0-9a-z_-]*` matches all current section names, see https://play.golang.org/p/2JytW6o-p_

Comment (Merovius) [reply manpages.go]: k fixed

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
I’ll leave it up to @cherti to decide whether to do another round of review or merge directly.

File: manpages.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var manpagesMatcher = regexp.MustCompile(`\b(\w+)\((\d[\da-z_-]*)\)(\W|$)`)
```
Comment (cherti) [inline manpages.go]: We should add a comment what this shall match, just that later readers of the code don't have to compile the Regex in their head to understand its purpose.

**Review by cherti — CHANGES_REQUESTED** [review]
Only one nitpick regarding the regex, otherwise this looks good to me

File: manpages.go (hunk 3 above)
Comment (cherti) [reply manpages.go]: Doesn't have to be an explicit explanation, just something like "shall match strings like "…" and extract the following parts"

Comment (chrko) [reply manpages.go]: Hm, I would just point the corresponding tests.

**Comment by Merovius** [comment]
@cherti PTAL

**Review by cherti — APPROVED** [review]

**Comment by cherti** [comment]
LGTM, thanks!

**Pull request merged**